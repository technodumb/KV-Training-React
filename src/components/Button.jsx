import React from "react";

const Button = ({ type = "button", children, onClick, isPrimary }) => {
    return (
        <button
            data-testid="button-test-id"
            type={type}
            onClick={onClick}
            className={"form-button " + (isPrimary ? "primary-button" : "")}
        >
            {children}
        </button>
    );
};

export default Button;
