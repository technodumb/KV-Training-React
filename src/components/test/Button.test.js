import { fireEvent, render } from "@testing-library/react";
import Button from "../Button";
import React from "react";

describe("Check if Button works properly", () => {
    test("Check if Button exists", () => {
        const { getByTestId } = render(<Button />);

        const element = getByTestId("button-test-id");
        expect(element).toBeTruthy();
    });

    test("Check if text displayed properly", () => {
        const text = "Click";
        const { getByText } = render(<Button>{text}</Button>);

        getByText(text);
    });

    test("Check if onClick is triggered", () => {
        const onClick = jest.fn();
        const { getByTestId } = render(<Button onClick={onClick} />);
        const element = getByTestId("button-test-id");

        fireEvent.click(element);
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test("Check if snapshot detects changes", () => {
        const onClick = jest.fn();
        const text = "Click";
        const { asFragment } = render(
            <Button onClick={onClick}>{text}</Button>
        );
        // const element = getByTestId("button-test-id");
        expect(asFragment()).toMatchSnapshot();
    });
});
