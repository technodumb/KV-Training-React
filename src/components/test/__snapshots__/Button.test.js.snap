// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Button works properly Check if snapshot detects changes 1`] = `
<DocumentFragment>
  <button
    class="form-button "
    data-testid="button-test-id"
    type="button"
  >
    Click
  </button>
</DocumentFragment>
`;
